jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -254,6 +254,14 @@ static long getTimeStamp(Request request)
     // TODO: see above.
     boolean isSecure();
 
+    /**
+     * Signal the intention to read the request content for the purposes of 100-Continues processing.
+     * @see HttpStream#willRead()
+     */
+    default void willRead()
+    {
+    }
```

Comment:
I think we can avoid adding this method, as it is only called from EEN, and we can call `writeInterim()` instead (perhaps have a `static write100()` utility method in `Response`).

The state about the 100 is kept in the `HttpStream` where we can easily decide whether to actually send it or not, so we won't need the extra methods in `Request` and `HttpStream`.

Reply:
@sbordet  So are you saying we change the semantic of `writeInterim(...)` to `sendInterimIfAppropriate(...)` ?   I .e. it only sends the interim after checking if it OK to send it.

I'll give it a go, but is it really, better than `willRead`?   Without a rename, it will be very confusing.

Reply:
No I'm saying to call `writeInterim()` as it is.

It will call `HttpStream.send()`.

There, we have all the state to figure out whether the interim is 100, whether we had the `Expect` header for 100, etc. and if the interim must really be written (or if it has already).

100 is a special case of interim as you can't send it more than once, it needs an `Expect` header, etc.
103, is very different in that regard and needs no state.

So the "if appropriate" is implicit for 100 because of what it is, and it does not apply to other interims.